compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1763,6 +1765,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         match_place: Place<'tcx>,
         test: &Test<'tcx>,
         mut candidates: &'b mut [&'c mut Candidate<'pat, 'tcx>],
+        mut catch_full_matched: impl FnMut(&Candidate<'_, 'tcx>, &MatchPair<'_, 'tcx>),
     ) -> (
         &'b mut [&'c mut Candidate<'pat, 'tcx>],
         FxIndexMap<TestBranch<'tcx>, Vec<&'b mut Candidate<'pat, 'tcx>>>,
```

Comment:
Return an extra map that records spans for the match pair that gave the first candidate, if it was `fully_matched`. You will need to add adequate comments to explain what this is doing.
```suggestion
        FxIndexMap<TestBranch<'tcx>, Vec<&'b mut Candidate<'pat, 'tcx>>>,
        FxIndexMap<TestBranch<'tcx>, Span>,
```